### PR TITLE
Feature rearrange storage

### PR DESCRIPTION
**Pull request by PrimaMateria**
Hi guys,
this is my first contribution to other github project and also first time I worked with react, so please be critical.
It is related to issue Rearrange storage #741.

Quick description:
From StorageItem extracted FolderList and FolderItem as separate components.
Wrapped them in the rect-sortable-hoc.
Implemented new data api method called reorderFolder.

I hope you will like it.
Cheers
Matus

### CODE REVIEW

**Comment by kazup01** [comment]
Thanks @PrimaMateria for your contribution!🎉
We will check it!

**Comment by kazup01** [comment]
Firstly, could you fix the CI error?

**Comment by PrimaMateria** [comment]
Here comes the screenshot:
screen shot 2017-09-08 at 14 29 44

Ok, I will later have look on the failed checks.

File: browser/main/lib/dataApi/reorderFolder.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  let targetStorage
+  try {
+    if (!_.isNumber(oldIndex)) throw new Error('oldIndex must be a number.')
+    if (!_.isNumber(newIndex)) throw new Error('newIndex must be a number.')
+
+    rawStorages = JSON.parse(localStorage.getItem('storages'))
+    if (!_.isArray(rawStorages)) throw new Error('Target storage doesn\'t exist.')
+
+    targetStorage = _.find(rawStorages, {key: storageKey})
+    if (targetStorage == null) throw new Error('Target storage doesn\'t exist.')
```
Comment (asmsuechan) [inline browser/main/lib/dataApi/reorderFolder.js]: Can you use `findStorage()`? https://github.com/BoostIO/Boostnote/blob/master/browser/lib/findStorage.js

File: tests/dataApi/reorderFolder-test.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+const test = require('ava')
```
Comment (asmsuechan) [inline tests/dataApi/reorderFolder-test.js]: Good :smile:

File: browser/main/lib/dataApi/reorderFolder.js (hunk 1 above)
Comment (PrimaMateria) [reply browser/main/lib/dataApi/reorderFolder.js]: Why is `findStorage` in `browser/lib` and not in `browser/main/lib/dataApi`?
Besides my `reorderFolder` I also updated `updateFolder`.
There was one more possible candidate `renameStorage`, but I didn't want to touch it now in the scope of this issue.

**Comment by PrimaMateria** [comment]
Tests are failing randomly. I put log into `beforeEach` and `after` for reorderFolder-test and moveNote-test and it seems like one test is clearing `localStorage` during the execution of other. 

This is log from the failed run:

```
reorderFolder setting items to localStorage
moveNote setting items to localStorage
moveNote clearing localStorage

   19 passed
   1 failed


   1. dataApi › reorderFolder-test › Reorder a folder
   failed with "Cannot read property 'key' of undefined"
      Test.fn (reorderFolder-test.js:43:58)
    _combinedTickCallback (internal/process/next_tick.js:141:11)
    process._tickCallback (internal/process/next_tick.js:180:9)
```

And these are 2 logs from successful run:
```
reorderFolder setting items to localStorage
/tmp/test/reorder-folder/boostnote.json
moveNote setting items to localStorage
reorderFolder clearing localStorage
/tmp/test/update-folder/boostnote.json
/tmp/test/create-folder/boostnote.json
moveNote clearing localStorage
```

```
/tmp/test/create-folder/boostnote.json
reorderFolder setting items to localStorage
/tmp/test/update-folder/boostnote.json
/tmp/test/reorder-folder/boostnote.json
reorderFolder clearing localStorage
moveNote setting items to localStorage
moveNote clearing localStorage
```

You can see that order is always different.

**Comment by asmsuechan** [comment]
> Tests are failing randomly.

I know it. But The fixing cost >>>>>> Push the restart button.

File: browser/main/lib/dataApi/reorderFolder.js (hunk 1 above)
Comment (asmsuechan) [reply browser/main/lib/dataApi/reorderFolder.js]: Because it just operates `localStorage`, does not operate files.

**Comment by PrimaMateria** [comment]
Hi guys, I am sorry, I don't know how the process works now. 
Is there something left for me to do?

**Comment by asmsuechan** [comment]
> Is there something left for me to do?

No, there's nothing to say about your PR. I left cross-OS checks for my review but I'm busy in these days so I could not take a time to do it. I'll work on this PR again in a week.

**Comment by asmsuechan** [comment]
Works fine! Thank you!

**Comment by kazup01** [comment]
Hi @PrimaMateria , I wrote about this PR in my article. Check it! https://medium.com/boostnote/thanks-to-your-contribution-boostnote-v0-8-16-has-been-released-783afea95ca3

**Comment by carugnom** [comment]
Sorry, I have version 0.11.9 installed and I don't see how to reorder the storages list.
Is this feature implemented ? If so, how do I use it ?

**Pull request merged**